float64[3] r
float64[4] q
float64[3] v
float64[3] ba
double tStamp;